### PR TITLE
Toggling fast scroll will actually make it work with no additional keypresses

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
```SUMMARY: Interface "Toggling fast scroll will actually make it work with no additional keypresses."```

#### Purpose of change
Using `Toggle fast scroll` option on overmap UI isn't making overmap scroll any faster at the moment until user manually changes `Move view offset` value in game options to something more than 1. And there is no information anywhere on this feature, except for [PR](https://github.com/CleverRaven/Cataclysm-DDA/pull/26444) that added Fast Scroll feature. Players might be confused (myself included, despite I was tracking that PR).
So I decided to add another option (independent of `Move view offset` option) to make overmap scroll faster by 5 squares by default when enabling Fast Scroll option.

#### Describe the solution
Added `FAST_SCROLL_OFFSET` option with default of 5 squares per keypress and make `Toggle fast scroll` option look for `FAST_SCROLL_OFFSET` value instead of `MOVE_VIEW_OFFSET`.

#### Describe alternatives you've considered
Not adding separate option, but then I thought that users might want to shift view on the game map by X squares and shift view on the overmap by Y squares. So I stopped on having two separate options.

### CODE REVIEW

**Comment by esotericist** [comment]
I've been meaning to do something about that myself, for basically all the exact reasons you specify here. It just kept not bubbling up to the top of my list.

I'm glad you got to it.

Will this also work for movement in the "look around" menu? It uses the same 'fast scroll' feature, linked to the `move_view_offset`?

**Comment by Night-Pryanik** [comment]
> Will this also work for movement in the "look around" menu? It uses the same 'fast scroll' feature, linked to the move_view_offset?

Yeah, it always worked like that.

**Comment by esotericist** [comment]
> > Will this also work for movement in the "look around" menu? It uses the same 'fast scroll' feature, linked to the move_view_offset?
> 
> Yeah, it always worked like that.

I'm aware it worked like that (since I've used the feature), I'm just verifying for my understanding if your change will apply to it too.

**Comment by Night-Pryanik** [comment]
> I'm just verifying for my understanding if your change will apply to it too.

No, it's not. Look around menu uses `MOVE_VIEW_OFFSET` option to determine by how many squares shift view. I didn't change anything on this feature. Scrolling in the overmap menu uses the newly added `FAST_SCROLL_OFFSET` option. They don't overlap.

**Comment by esotericist** [comment]
> > I'm just verifying for my understanding if your change will apply to it too.
> 
> No, it's not. Look around menu uses `MOVE_VIEW_OFFSET` option to determine by how many squares shift view. I didn't change anything on this feature. Scrolling in the overmap menu uses the newly added `FAST_SCROLL_OFFSET` option. They don't overlap.

Then I would request that you also address the look around menu while you're at it, because the exact same problem applies to it as applies to the overmap. 

To be clear: I'm not talking about shifting the offset of the main view (with `H`, `J`, etc), but specifically using `x` and then moving the cursor. This supports using `f` for fast movement exactly the same as the overmap, including the fact it does nothing until you change the `MOVE_VIEW_OFFSET`

**Comment by Night-Pryanik** [comment]
> using x and then moving the cursor. This supports using f for fast movement exactly the same as the overmap, including the fact it does nothing until you change the MOVE_VIEW_OFFSET

Whoa, I didn't even know that!
So what do you want specifically? To make fast scroll while looking around by 'x' look on `FAST_SCROLL_OFFSET` instead of a `MOVE_VIEW_OFFSET` or what?

**Comment by esotericist** [comment]
> > using x and then moving the cursor. This supports using f for fast movement exactly the same as the overmap, including the fact it does nothing until you change the MOVE_VIEW_OFFSET
> 
> Whoa, I didn't even know that!
> So what do you want specifically? To make fast scroll while looking around by 'x' look on `FAST_SCROLL_OFFSET` instead of a `MOVE_VIEW_OFFSET` or what?

Yes, that is what I want. `MOVE_VIEW_OFFSET` should, in my opinion, be exclusively changing how far you adjust your center-of-screen offset with `H`, `J`, etc in the main game view.

`x` screen when you toggle on fast movement with `f` should behave the same as overmap fast movement, including using the new configuration setting. Without `f` for fast movement toggled on, cursor movement in the look_around `x` screen should continue to behave as it does now (moving one tile at a time)

**Comment by AMurkin** [comment]
What if we show a message asking user to input desired value if it's equal to 1 when he toggles the fast scroll?

**Comment by Night-Pryanik** [comment]
Jenkins rebuild

**Comment by Night-Pryanik** [comment]
@esotericist Done as you requested.

**Comment by Night-Pryanik** [comment]
> What if we show a message asking user to input desired value if it's equal to 1 when he toggles the fast scroll?

I think that with new fast scroll defaults of 5 it is no longer needed.

File: src/options.cpp [hunk 1]
```diff
@@ -1418,6 +1418,11 @@ void options_manager::add_options_interface()
          1, 50, 1
        );
 
+    add( "FAST_SCROLL_OFFSET", "interface", translate_marker( "Overmap fast scroll offset" ),
+        translate_marker( "With Fast Scroll option enabled, shift view on the overmap and while looking around by this many squares per keypress." ),
```
Comment (ZhilkinSerg) [inline src/options.cpp]:
```suggestion
         translate_marker( "With Fast Scroll option enabled, shift view on the overmap and while looking around by this many squares per keypress." ),
```

File: src/options.cpp [hunk 2]
```diff
@@ -1418,6 +1418,11 @@ void options_manager::add_options_interface()
...
+        1, 50, 5
```
Comment (ZhilkinSerg) [inline src/options.cpp]:
```suggestion
         1, 50, 5
```

File: src/options.cpp [hunk 3]
```diff
@@ -1418,6 +1418,11 @@ void options_manager::add_options_interface()
...
+    );
```
Comment (ZhilkinSerg) [inline src/options.cpp]:
```suggestion
       );
```

**Comment by Night-Pryanik** [comment]
Added a hint to the Look Around menu about fast scroll option. It's green when disabled and light green when enabled (analogue to cyan and magenta colored hints on overmap menu). This replaced old "F" symbolic indicator when fast scroll is enabled.

изображение
изображение

**Comment by esotericist** [comment]
> Added a hint to the Look Around menu about fast scroll option. It's green when disabled and light green when enabled (analogue to cyan and magenta colored hints on overmap menu). This replaced old "F" symbolic indicator when fast scroll is enabled.

That is clever and genuinely a good idea. The discoverability on this feature was really poor.

**Pull request merged**